### ISSUE TITLE
Have `.where` be a noop is the condition passed in is a empty object

### ISSUE THREAD

**Issue by joelwat** [benbotto/node-data-mapper#1]
I have a retrieve function that sometimes return everything and sometimes selectively. It would be nice to not to have to break up my function chain and put an if statement in there. Currently I would have to do something like this.
``` javascript
class ProjectsDao {
    retrieve(cond = {}, params = {}) {
      const query = db.dataContext
        .from('projects AS p');

      if (Object.keys(cond).length !== 0 || cond.constructor !== Object) {
        query.where(cond, params);
      }

      query
        .select()
        .execute()
        .then((rVal) => {
          if (!rVal.projects.length) {
            return deferred.reject(
              new NotFoundError('No projects were found.')
            );
          }

          return rVal.projects;
        });
    }
  }
```
I would like it to look like this.
``` javascript
class ProjectsDao {
    retrieve(cond = {}, params = {}) {
      return db.dataContext
        .from('projects AS p')
        .where(cond, params)
        .select()
        .execute()
        .then((rVal) => {
          if (!rVal.projects.length) {
            return deferred.reject(
              new NotFoundError('No projects were found.')
            );
          }

          return rVal.projects;
        });
    }
  }
```
I think it makes my function simpler and look cleaner.

**Comment by benbotto** [comment]
Pull request looks fine.  It seems like passing in null/undefined for `where()`'s `cond` parameter should work the same way, i.e. not add a where clause.  Take a look at this commit, which has your pull request as well as support for undefined/null: https://github.com/benbotto/node-data-mapper/commit/0b0559095733e11c073f5c294f66d5383573a134

If that looks good to you, I'll merge up to 1.1.x and publish the new version.

**Comment by joelwat** [comment]
You're right. It should handle undefined/null in the same manner.

You modifications to the pull request look good to me. Please merge it and publish.

**Comment by benbotto** [comment]
Okay, published as 1.1.26.  Thanks.
